### ISSUE TITLE
TextBox Focused VisualState rendering incorrectly on WASM

### ISSUE THREAD

**Issue by peternary** [unoplatform/uno#1085]
## Current behavior

The Textbox VisualState for Focused seems to draw with a white background regardless of any custom VisualState

In a custom TextBox with a border this background draws over the border on focus, making the border disappear

## Expected behavior

Control should draw in same fashion as on UWP

## How to reproduce it (as minimally and precisely as possible)

[ControlStyleSample.zip](https://github.com/nventive/Uno/files/3297534/ControlStyleSample.zip)

Build and test for UWP, clicking off and on control to observe change in visual style.  Build and test for WASM, notice that border appears to vanish in the area around the textbox.

## Environment

Nuget Package:
Uno.UI
Package Version(s):
1.45.0-dev.1828
Affected platform(s):
- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] Windows
- [ ] Build tasks

Visual Studio
- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins
- [ ] Resharper (version: )

**Comment by carldebilly** [comment]
I checked your sample and I found this...

![image](https://user-images.githubusercontent.com/4174207/59614822-7ec60180-90ef-11e9-934b-f1a46c0a3620.png)

Is it what you mean by _borders appears to vanish_?

**Comment by peternary** [comment]
Correct

**Comment by davidjohnoliver** [comment]
This turns out to be because the `Style` isn't applied to the `TextBox`, this will be fixed by #1157 .

Note that I had to remove the following lines from the visual states setters in the style after the fix:
```xaml
<ObjectAnimationUsingKeyFrames Storyboard.TargetName="HeaderContentPresenter"
Storyboard.TargetProperty="Foreground">
	<DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource SystemControlDisabledBaseMediumLowBrush}" />
</ObjectAnimationUsingKeyFrames>
```

The compiler complains because there's no view in the template named `HeaderContentPresenter`, Uno is currently stricter than UWP in this regard.
